lib/library.rb

```diff
@@ -0,0 +1,64 @@
+require 'yaml'
+require 'date'
+require './lib/person.rb'
+
+class Library
+  attr_accessor :books, :title, :available, :renter, :return_date
+
+  def initialize(books, title, available, renter, return_date)
+    @books = books
+    @title = title
+    @available = available
+    @renter = name
+    @return_date = date.today.next_month
+
+
+  end
+
+  def list_available
+    books.select { |obj| obj[:available] == true }
+  end
+
+  def list_all
+    YAML.load_file('./lib/books.yml')
+  end
+
+  def search_books
+    def search_books(title)
+  book = books.select { |obj| obj[:item][:title] == title}
+  if book[:available] == false
+    'Not available'
+  else
+    book
+  end
+end
```

Comment:
Good method. I'm worried where the values on `books` that you iterate over is being set. The way I look at it, theres an big risk that it's empty and you get a NilClass exception. 

Please mind code formatting.